lib/app/console_app.rb

```diff
@@ -9,9 +9,76 @@
 $LOAD_PATH << '../'
 
 require 'app/config'
+require 'connectors/base/registry'
 
 module App
   module ConsoleApp
-    puts 'Hello Connectors 3.0'
+    extend self
+
+    INDEX_NAME_REGEXP = /[a-zA-Z]+[\d_\-a-zA-Z]*/
+
+    def start_sync
+      puts 'Please enter index name for data ingestion. Use only letters, underscored and dashes.'
+      index_name = gets.chomp.strip
+      unless INDEX_NAME_REGEXP.match?(index_name)
+        puts "Index name #{index_name} contains symbols that aren't allowed!"
+        return
+      end
+
+      connector = select_connector({ :index_name => index_name })
+
+      puts 'Starting content sync...'
+      connector.sync_content
+    end
+
+    def show_status
+      connector = select_connector
+
+      puts 'Checking status...'
+      puts connector.source_status
+      puts
+    end
+
+    def read_command
+      puts 'Please enter a command. Available options:'
+      puts '- sync - start synchronization'
+      puts '- status - check the status of a third-party service'
+      puts '- exit - end the program'
+      puts
+      gets.chomp.to_sym
+    end
+
+    def select_connector(params = {})
+      puts 'Registered connectors:'
+      connectors = registry.registered_connectors
+      connectors.each_with_index { |name, index| puts "#{index} - #{name}" }
+      puts
+      puts 'Please enter the number of the connector you want:'
+      order = gets.chomp.to_i
+      registry.connector(connectors[order], params)
+    end
+
+    def registry
+      @registry = Connectors::Base::REGISTRY
+    end
+
+    puts 'Hello Connectors 3.0!'
+
+    while true
+      command = read_command
+      if command == :sync
```

Comment:
looks like a good candidate for a `case` statement

Reply:
yeah there were just 2 commands initially, so...
I've redone as case now.